vars/buildPlugin.groovy

```diff
@@ -48,224 +48,227 @@ def call(Map params = [:]) {
     }
 
     tasks[stageIdentifier] = {
-      node(label) {
-        try {
-          timeout(timeoutValue) {
-            boolean isMaven
-            // Archive artifacts once with pom declared baseline
-            boolean doArchiveArtifacts = !jenkinsVersion && !archivedArtifacts
-            if (doArchiveArtifacts) {
-              archivedArtifacts = true
-            }
-            boolean incrementals // cf. JEP-305
-
-            stage("Checkout (${stageIdentifier})") {
-              infra.checkoutSCM(repo)
-              isMaven = fileExists('pom.xml')
-              incrementals = fileExists('.mvn/extensions.xml') &&
-                  readFile('.mvn/extensions.xml').contains('git-changelist-maven-extension')
-              final String gitUnavailableMessage = '[buildPlugin] Git CLI may not be available'
-              withEnv(["GITUNAVAILABLEMESSAGE=${gitUnavailableMessage}"]) {
-                if (incrementals) { // Incrementals needs 'git status -s' to be empty at start of job
-                  if (isUnix()) {
-                    sh 'git clean -xffd || echo "$GITUNAVAILABLEMESSAGE"'
-                  } else {
-                    bat 'git clean -xffd || echo %GITUNAVAILABLEMESSAGE%'
-                  }
-                }
+      // TODO use kubernetesAgent() if we know we are using a pod
```

Comment:
I have not settled on the right DX here. Currently `kubernetesAgent` matches only when used on K8s agents. So if you use `conditions: [kubernetesAgent()]` then you will get no retries of non-K8s (e.g., Docker VM) `node` blocks, which seems like the expected behavior. And `conditions: [agent(), kubernetesAgent()]` is logically the same as ``conditions: [agent()]`: matches whenever the node is lost, disregarding K8s details (like genuine problems with the pod definition). I guess there could be an option like `conditions: [kubernetesAgent(handleNonKubernetes: true)]` which would fall back to generic `agent()` behavior when given a non-K8s agent, which is what we would ideally use here where we cannot easily tell what sort of agent is being used.

Reply:
so if you were to use:
```
conditions: [agent(), kubernetesAgent()]
```
on a non kubernetes agent (because you don't know when you'll be on kubernetes and when not) it won't cause a problem?

Reply:
It would not cause a problem, it would just behave identically to

```groovy
conditions: [agent()]
```

Same for

```groovy
conditions: [kubernetesAgent(), agent()]
```

Reply:
In other words, when used with a K8s agent, it would retry `node` even though the `kubernetes` plugin knows that in this case the problem was _not_ an infrastructure outage: something like `OOMKilled` (bad memory limit request) or `DeadlineExceeded` (e.g., missing container image) that is really the developer’s concern.

Reply:
⇒ https://github.com/jenkinsci/kubernetes-plugin/pull/1190/commits/c38fe0fabf27d0583c9b546aa701dc075245e72d